internal/services/groups/groups_data_source.go

```diff
@@ -87,8 +105,7 @@ func groupsDataSourceRead(ctx context.Context, d *schema.ResourceData, meta inte
 		if len(*result) == 0 {
 			return tf.ErrorDiagPathF(err, "return_all", "No groups found")
 		}
-
-		groups = append(groups, *result...)
+		groups = filterResults(securityEnabled, mailEnabled, result)
```

Comment:
Could we compose odata filters for this too, to minimise the number of unrelated results going over the wire? e.g. `securityEnabled eq true` and/or `mailEnabled eq true`

Reply:
I may be being stupid and not understanding odata querying - But would this not lead to something like:

`			if securityEnabled && mailEnabled{
				query := odata.Query{
					Filter: fmt.Sprintf("displayName eq '%s', securityEnabled eq true, mailEnabled eq true", displayName),
				}
			} else if securityEnabled {
				query := odata.Query{
					Filter: fmt.Sprintf("displayName eq '%s', securityEnabled eq true", displayName),
				}
			} else if mailEnabled {
				query := odata.Query{
					Filter: fmt.Sprintf("displayName eq '%s', mailEnabled eq true", displayName),
				}
			} else {
				query := odata.Query{
					Filter: fmt.Sprintf("displayName eq '%s'", displayName),
				}
			}`